scrapy_poet/injection.py

```diff
@@ -294,6 +300,11 @@ def build_instances_from_providers(
                 )
             instances.update(objs_by_type)
 
+            if self.weak_cache.get(request):
+                self.weak_cache[request].update(objs_by_type)
+            else:
+                self.weak_cache[request] = objs_by_type
```

Comment:
So this fills the cache. Where is it used?

Reply:
Sorry, just updated the PR's description to reflect this. It's primarily used now in `ZyteApiProvider` and replaces the weakref cache in it as well: https://github.com/scrapy-plugins/scrapy-zyte-api/pull/161/commits/3cb22901c762aa2f0503707312a2ebabcefc42bc